src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -381,54 +388,71 @@ public void Stake(WalletSecret walletSecret)
         {
             this.logger.LogTrace("()");
 
-            if (Interlocked.CompareExchange(ref this.stakeProgressFlag, StakeInProgress, StakeNotInProgress) == StakeInProgress)
+            if (Interlocked.CompareExchange(
+                    ref this.stakeStateFlag,
+                    (int)CurrentState.Executing,
+                    (int)CurrentState.Idle) == (int)CurrentState.Executing)
             {
                 this.logger.LogTrace("(-)[ALREADY_MINING]");
                 return;
             }
 
-            this.rpcGetStakingInfoModel.Enabled = true;
-            this.stakeCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(new[] { this.nodeLifetime.ApplicationStopping });
-
-            this.stakingLoop = this.asyncLoopFactory.Run("PosMining.Stake", async token =>
+            if (Interlocked.CompareExchange(
+                    ref this.stopStakingStateFlag,
+                    (int)CurrentState.Idle,
+                    (int)CurrentState.Executing) == (int)CurrentState.Executing)
             {
-                this.logger.LogTrace("()");
+                this.logger.LogTrace("(-)[MINING_STOPPING]");
+                return;
+            }
 
-                try
-                {
-                    await this.GenerateBlocksAsync(walletSecret).ConfigureAwait(false);
-                }
-                catch (OperationCanceledException)
-                {
-                    // Application stopping, nothing to do as the loop will be stopped.
-                }
-                catch (MinerException me)
-                {
-                    // Miner exceptions should be ignored. It means that the miner
-                    // possibly mined a block that was not accepted by peers or is even invalid,
-                    // but it should not halted the staking operation.
-                    this.logger.LogDebug("Miner exception occurred in miner loop: {0}", me.ToString());
-                    this.rpcGetStakingInfoModel.Errors = me.Message;
-                }
-                catch (ConsensusErrorException cee)
-                {
-                    // All consensus exceptions should be ignored. It means that the miner
-                    // run into problems while constructing block or verifying it
-                    // but it should not halted the staking operation.
-                    this.logger.LogDebug("Consensus error exception occurred in miner loop: {0}", cee.ToString());
-                    this.rpcGetStakingInfoModel.Errors = cee.Message;
-                }
-                catch
+            this.rpcGetStakingInfoModel.Enabled = true;
+            this.stakeCancellationTokenSource =
```

Comment:
Here I would argue that it definitely does not belong onto separated lines. Probably your IDE setting or resharper is set with quite a low number of characters on one line. Please consider week your settings.

Reply:
Agree, fixed. It was autoformatting doing it.